packages/kbn-eslint-plugin-eslint/rules/module_migration.js

```diff
@@ -95,7 +101,32 @@ module.exports = {
     ],
   },
   create: (context) => {
-    const mappings = context.options[0];
+    const _mappings = context.options[0];
+    const filename = path.relative(KIBANA_ROOT, context.getFilename());
+
+    const mappings = _mappings.filter((mapping) => {
+      if (mapping.include || mapping.exclude) {
+        if (mapping.include) {
+          let include = false;
+          mapping.include.forEach((rgx) => {
+            if (rgx.test(filename)) {
+              include = true;
+            }
+          });
+          return include;
+        }
+        if (mapping.exclude) {
+          let include = true;
+          mapping.exclude.forEach((rgx) => {
+            if (rgx.test(filename)) {
+              include = false;
+            }
+          });
+          return include;
+        }
+      }
+      return true;
+    });
```

Comment:
This way this is written `include` options will prevent any `exclude` options from being considered, and I think the use of `Array.some()` would be more appropriate. In this alternate approach `exclude` rules are checked first and exclude the mapping if any exclude rule matches. Then `include` rules are checked, only using the mapping if some rule matches. In other cases mapping rules are used.

```suggestion
    const filename = path.relative(KIBANA_ROOT, context.getFilename());

    const mappings = context.options[0].filter((mapping) => {
      // exclude mapping rule if it is explicitly excluded from this file
      if (mapping.exclude && mapping.exclude.some((p) => p.test(filename))) {
        return false;
      }

      // if this mapping rule is only included in specific files, optionally include it
      if (mapping.include) {
        return mapping.include.some((p) => p.test(filename));
      }

      // include all mapping rules by default
      return true;
    });
```